xl-v1-metadata.go

```diff
@@ -259,39 +282,18 @@ func (xl xlObjects) renameXLMetadata(srcBucket, srcPrefix, dstBucket, dstPrefix
 	// Wait for all the routines.
 	wg.Wait()
 
-	// Gather err count.
-	var errCount = 0
-	for _, err := range mErrs {
-		if err == nil {
-			continue
+	if !isQuorum(mErrs, xl.writeQuorum) {
+		if isQuorum(mErrs, xl.readQuorum) {
+			return nil
 		}
-		errCount++
+		xl.undoRenameXLMetadata(srcBucket, srcPrefix, dstBucket, dstPrefix, mErrs)
```

Comment:
> renameXLMetadata() source is always a tmp location so undoRenameXLMetadata to rename it back to a tmp file will not be useful

It would be since 'tmp/' location can be cleaned naturally without issues, any dangling files would be an issue to detect and cleanup later.